### DISCUSSION TITLE
Filter for sub shapes

### DISCUSSION THREAD

**Discussion by DocAce** [jrouwe/JoltPhysics#167] [Q&A] [answered]
I have a physics body with a compound collision shape and would like to be able to ignore some (transparent) child shapes for certain raycasts (e.g. lens flares, visibility checks). I would prefer to keep everything in a single body.
The child shape already has the required data in a custom DecoratedShape, but the BodyFilter is of course only called to filter entire bodies. Would it be possible (or rather: recommended) to either extend the BodyFilter or to add some kind of SubShapeFilter class?

**Comment by jrouwe** [comment, marked as answer]
NarrowPhaseQuery::CastShape / TransformedShape::CastShape already offers the interface to pass a ShapeFilter class to do what you're asking for. I didn't need a shape filter on CastRay/CollidePoint/CollideShape so I never implemented it. I added it for CastRay as well now in #168. Try it out.

**Reply by DocAce** [reply]
Thanks a ton, I will try it asap!
Edit: it works, thanks again :)

**Reply by jrouwe** [reply]
Change has been merged to master.
